lightning/src/ln/async_signer_tests.rs

```diff
@@ -329,6 +330,242 @@ fn test_async_commitment_signature_for_peer_disconnect() {
 	}
 }
 
+#[test]
+fn test_async_commitment_signature_ordering_reestablish() {
+	do_test_async_commitment_signature_ordering(false, false, false);
+}
+
+#[test]
+fn test_async_commitment_signature_ordering_monitor_restored() {
+	do_test_async_commitment_signature_ordering(true, false, false);
+}
+
+#[test]
+fn test_async_commitment_signature_ordering_monitor_restored_signer_restored_before_completion() {
+	// This tests that if we were pending a monitor update completion across a restart,
+	// and needed to send a CS then RAA, that if our signer is unblocked before the monitor
+	// update completes, then we don't send duplicate messages upon calling `signer_unblocked`
+	// after the monitor update completes.
+	do_test_async_commitment_signature_ordering(true, true, false);
+}
+
+#[test]
+fn test_async_commitment_signature_ordering_signer_restored_before_reestablish() {
+	// This tests that if we tried to send a commitment_signed, but our signer was blocked,
+	// if we disconnect, reconnect, unblock the signer, then handle channel_reestablish,
+	// that we don't send duplicate messages upon calling `signer_unblocked`.
+	do_test_async_commitment_signature_ordering(false, false, true);
+}
+
+fn do_test_async_commitment_signature_ordering(monitor_update_failure: bool,
+	enable_signer_before_monitor_completion: bool, enable_signer_before_reestablish: bool) {
```

Comment:
after the latest additional tests this helper is a bit unwieldy, not sure the best way to address it. I could copy this test 4 times and get rid of the conditionals, but it adds a lot of filler...

Reply:
yaknow actually i realized for what these are testing it isn't necessary to include them in this test (they don't care about the CS-RAA ordering, just that the CS signer pending flag is cleared at the right time), i'll just move them to their own separate tests tomorrow

Reply:
okay, fixed this, moved them into the peer disconnection test